src/main/java/Storage.java

```diff
@@ -0,0 +1,117 @@
+import java.io.*;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Encapsulates the operations related to the I/O of the log file.
+ */
+public class Storage {
+    String pathname;
+    List<Task> data = new ArrayList<>();
+
+    /**
+     * Constructs a storage object.
+     * @param path the file path.
+     */
+    public Storage(String path) {
+        this.pathname = path;
+    }
+
+    /**
+     * Reads data from the log file.
+     * @return data that represented by a list of Task objects.
+     * @throws IOException Handle the exception when reading files.
+     */
+    public List<Task> readFile() throws IOException {
+        String pathname = "./log.txt";
+        File filename = new File(pathname);
+        InputStreamReader reader = new InputStreamReader(new FileInputStream(filename));
+        BufferedReader br = new BufferedReader(reader);
+        String line;
+        line = br.readLine();
+        while (line != null) {
+            char type = line.charAt(8);
+            boolean isDone = line.charAt(11) == '✓';
+            if(type == 'T') {
+                Todo t = new Todo(line.substring(14));
+                if(isDone) {
+                    t.done();
+                }
+                data.add(t);
+            } else if(type == 'D') {
+                String[] parts = line.substring(5).split(" ");
+                String description = "";
+                int length = parts.length;
+                for(int i = 1; i < length - 5; i++) {
+                    description += parts[i] + " ";
+                }
+                Deadline d = new Deadline(description, LocalDateTime.of(Integer.parseInt(parts[length - 2]),
+                        getMonth(parts[length - 4]), Integer.parseInt(parts[length - 3]),
+                        Integer.parseInt(parts[length - 1].split(":")[0]),
+                        Integer.parseInt(parts[length - 1].split(":")[1].substring(
+                                0, parts[length - 1].split(":")[1].length() - 1))));
```

Comment:
Could consider abstracting our each arguments and put them in a variable to make your code more readable.